.github/actions/restore_dbt_cache/action.yaml

```diff
@@ -41,16 +41,26 @@ runs:
             echo "cache-matched-key=$KEY"
           } >> $GITHUB_OUTPUT
         else
-          echo "Did not find exact match for cache key, checking fallback"
-          if aws s3api head-object --bucket "$BUCKET" --key "$RESTORE_KEY/manifest.json"; then
-            echo "Cache hit: Found fallback match"
-            {
-              echo "cache-hit=true";
-              echo "exact-match=false";
-              echo "cache-matched-key=$RESTORE_KEY"
-            } >> $GITHUB_OUTPUT
+          echo "Did not find exact match for cache key"
+          if [ ! -z "$RESTORE_KEY" ]; then
+            echo "Checking for fallback cache via restore-key"
+            if aws s3api head-object --bucket "$BUCKET" --key "$RESTORE_KEY/manifest.json"; then
+              echo "Cache hit: Found fallback match"
+              {
+                echo "cache-hit=true";
+                echo "exact-match=false";
+                echo "cache-matched-key=$RESTORE_KEY"
+              } >> $GITHUB_OUTPUT
+            else
+              echo "Cache miss: Did not find fallback match for cache key"
+              {
+                echo "cache-hit=false";
+                echo "exact-match=false";
+                echo "cache-matched-key=''";
+              } >> $GITHUB_OUTPUT
+            fi
```

Comment:
This block has not changed, it was simply indented further and gated to match the specific case where an exact `key` match was not found and a `restore-key` fallback is configured. Previously, `restore-key` was required, so this was the only possible logic that could be executed.